sound/soc/sof/ipc.c

```diff
@@ -537,8 +537,10 @@ static void ipc_period_elapsed(struct snd_sof_dev *sdev, u32 msg_id)
 
 	memcpy(&stream->posn, &posn, sizeof(posn));
 
-	/* only inform ALSA for period_wakeup mode */
-	if (!stream->substream->runtime->no_period_wakeup)
+	if (spcm->pcm.compress)
+		snd_sof_compr_fragment_elapsed(stream->cstream);
```

Comment:
Would it be too paranoid to test for `stream->cstream != NULL` also or just:
```
- 	if (spcm->pcm.compress)
+ 	if (stream->cstream)
```
The `stream->cstream != NULL` can be within `snd_sof_compr_fragment_elapsed()`

Reply:
Ok, we can add the check for stream->cstream != NULL inside snd_sof_compr_fragment_elapsed() for sure.

About making SND_SOC_SOF_COMPRESS a non user visible option I have no strong reason against. But this will force platforms to have this always compiled in even if maybe some users do not need it.

Reply:
When SOF gains the support for compressed then it is mostly going to be a topology change to have compressed support or not, no?
Can a user swap between topologies with and without compressed streams but using the same kernel?

Reply:
@ujfalusi exactly. The topology decides if the sound card is for compress or PCM. 

See 5d5d676813e1f8cd070f451fc7336219ed40abcc

We are already using this in our internal tree. Using the same kernel we load different topology for compress / PCM topologies.

Reply:
Well, it's equally problematic if
a) the user does not select the compressed stuff and the topology relies on it or
b) the platform does not select the compressed stuff and the topology relies on it

In both cases, we've got to have a clear error message that says 'invalid topology'

I personally think a platform decision is better, e.g. for Intel we have no plans on enabling compressed playback on 'legacy' platforms. It would be feasible but not a priority for us. So indicating what platforms do explicitly support compressed stuff would be less error-prone, or at least not set the expectation that this capability is supported on all SOF-based devices.

In case the platform allows for compressed stuff, the topology does not use the compressed path, then a module is not used, that's no big deal.

Reply:
Agree. We made SND_SOC_SOF_COMPRESS automatically selected only on designated platforms.

Reply:
> Would it be too paranoid to test for `stream->cstream != NULL` also or just:
> 
> ```
> - 	if (spcm->pcm.compress)
> + 	if (stream->cstream)
> ```
> 
> The `stream->cstream != NULL` can be within `snd_sof_compr_fragment_elapsed()`

We added `stream->cstream == NULL` check in `snd_sof_compr_fragment_elapsed()`